### PR TITLE
auto/manual init w/ single bundle, BYO root element 🎉

### PR DESCRIPTION
**Pull request by erquhart**

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 9bb1a81e4d822fd4309f8cc9fb90685eb4f930bc

https://deploy-preview-1173--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 9bb1a81e4d822fd4309f8cc9fb90685eb4f930bc

https://deploy-preview-1173--cms-demo.netlify.com

File: src/bootstrap.js [hunk 1]
```diff
@@ -23,11 +24,25 @@ function bootstrap(opts = {}) {
...
+    if (existingRoot) {
+      return existingRoot;
+    }
+
+    /**
+     * If no existing root, create and return a new root.
+     */
+    const newRoot = document.createElement('div');
+    newRoot.id = ROOT_ID;
+    document.body.appendChild(el);
```
Comment (talves) [inline src/bootstrap.js]: needs to be 
`document.body.appendChild(newRoot);`

**Comment by erquhart** [comment]
@talves after you came up with the idea of using a DOM element to trigger manual init, I took a second look at the whole thing and added a commit to really simplify everything by separating manual init and using a custom mount element. We can just use a global flag for manual init so that a custom element is optional.

Using a single bundle removes the risk of dual init, so we don't need those warnings anymore. Removing that apparatus also made resetting as simple as rerunning the init function, so it's pretty much gloves off if you use manual init, no safety rails :)

Let me know what you think.

**Comment by talves** [comment]
@erquhart I really like the simplification, but it introduced a side affect in the create-react-app test.

I will take another look to see why and try to get back to you soon

**Comment by erquhart** [comment]
Testing now, looks like there are a few errors.

**Comment by erquhart** [comment]
Fixed.

File: src/bootstrap.js [hunk 2]
```diff
@@ -13,7 +13,16 @@ import 'EditorWidgets';
...
+const ROOT_ID = 'nc-root';
+
 function bootstrap(opts = {}) {
+  /**
+   * Error and return if this function was already called.
+   */
+  if (document.getElementById(ROOT_ID)) {
+    console.error('Bootstrap attempted, but Netlify CMS is already initialized!');
+    return;
+  }
```
[the remark this answers is not in the document]
Comment (erquhart) [reply src/bootstrap.js]: I think you were looking at an old commit, this was removed.

**Comment by talves** [comment]
Ok, all tested and working as expected. 👍 
***NOTE:*** Make sure to use the correct global flag 🤣

**Pull request merged**